pkg/sql/pgwire/server.go

```diff
@@ -483,6 +487,22 @@ func (s *Server) ServeConn(ctx context.Context, conn net.Conn, socketType Socket
 	ctx, draining, onCloseFn := s.registerConn(ctx)
 	defer onCloseFn()
 
+	// Reject the connection if there are too many connections already.
+	// Note that we do not carve out an exception for `root` or admin
+	// users here: the max is per listener, and admin connections can
+	// be established using a secondary admin-only listener with no max
+	// conn limit instead.
+	// See: https://github.com/cockroachdb/cockroach/issues/44842
+	// See: https://github.com/cockroachdb/cockroach/issues/51453
+	if numConns := s.metrics.Conns.Value(); socketType == SocketTCP && s.maxConns > 0 && numConns > int64(s.maxConns) {
```

Comment:
Oh I see, this is actually limiting the total number of SQL connections, but it only checks during TCP. So a server could be full of only unix conns and then a single TCP connection would be rejected. Ok, if that's the intention then I rescind my rename comment above.